GrandCentralBoard/Utilities/GoogleAPIDataProvider.swift

```diff
@@ -0,0 +1,78 @@
+//
+//  GoogleAPIDataProvider.swift
+//  GrandCentralBoard
+//
+//  Created by Michał Laskowski on 06.04.2016.
+//  Copyright © 2016 Oktawian Chojnacki. All rights reserved.
+//
+
+import Alamofire
+import Operations
+
+enum APIDataError : ErrorType {
+    case AuthorizationError
+    case UnderlyingError(NSError)
+}
+
+final class GoogleAPIDataProvider {
+
+    private let tokenProvider: OAuthTokenProvider
+    private var accessToken: AccessToken?
+
+    private let networkRequestManager: NetworkRequestManager
+
+    private let operationQueue = OperationQueue()
+
+    init(tokenProvider: OAuthTokenProvider, networkRequestManager: NetworkRequestManager = Manager()) {
+        self.tokenProvider = tokenProvider
+        self.networkRequestManager = networkRequestManager
+    }
+
+    private func refreshTokenOperation() -> Operation {
+        let refreshTokenOperation = BlockOperation(block: { [weak self] (continueWithError) in
+            guard let strongSelf = self else {
+                continueWithError(error: nil)
+                return
+            }
+
+            strongSelf.tokenProvider.accessTokenFromRefreshToken({ result in
+                switch result {
+                case .Failure(let error):
+                    continueWithError(error: error)
+                case .Success(let value):
+                    self?.accessToken = value
```

Comment:
I tend to keep self alive in completion blocks for finite operations, but that's debatable.
What you said makes sense, and I'm ok with that. Just wanted to point it out and make sure it wasn't a mistake.